lib/index.js

```diff
@@ -427,6 +487,37 @@ api.checkPermission = function(actor, permission, options, callback) {
     });
 };
 
+function _checkMembershipUpdate(actor, identity, groups, callback) {
+  // EXTERNAL: check to make sure IDENTITY_UPDATE_MEMBERSHIP permissions
+  //           exists for actor
+  // for each group in groups
+  //  check to make sure group exists
+  //  check to make sure group is a type: Group
+  //  x check to make sure owner is actor OR
+  //      actor has IDENTITY_UPDATE_MEMBERSHIP permission on resource??
+  async.each(groups, (group, callback) => {
+    async.auto({
+      get: callback => {
+        const query = {
+          id: database.hash(group),
+          'identity.type': 'Group',
+          'identity.sysStatus': 'active'
+        };
+        database.collections.identity.findOne(query, {
```

Comment:
One downside to doing this is that we can't represent identities that are members of groups that we don't store on our own system. We could allow actors to set `memberOf` for groups that are either found and they have permission to join ... or that are `NotFound` ... if we're ok with the other consequences of doing so.